### DISCUSSION TITLE
UI bug with Copy the full SHA button on the branch compare page

### DISCUSSION THREAD

**Discussion by kj800x** [community/community#7954] [Other Feature Feedback, Questions, & Ideas]
I discovered a bug on the branch compare page of GitHub. One such page is https://github.com/kj800x/nulinks/compare/v0.3.9...master.

When you click on the button to "Copy the full SHA", it appears that the whole button is clickable but the SHA is only copied if you click on the actual icon inside the button. This leads to a very inconsistent experience, since most users would expect to be able to click anywhere in the button for the feature to work..

I've attached a GIF demonstrating the problem. I have tested and noticed that the SHA will not be copied unless I get the green checkmark. You can see that I click various places inside the button and I do not get the green checkmark (and the SHA is not copied) unless I click directly on the copy icon, which is not typically how buttons work.

![Kapture 2021-11-20 at 20 46 24](https://user-images.githubusercontent.com/1657142/142746657-22987586-235b-42da-afd3-840fb6d1c31b.gif)

I have tested and am able to reproduce the issue consistently on both Chrome and Firefox on Linux and Mac.

Thank you and I hope this bug report helps. Please let me know if there is somewhere else I should submit it!

**Comment by Intrepidd** [comment]
Yes !!!!

It took me several days to understand why copying the sha only worked 50% of the time

Huge QOL improvement :)
